### PR TITLE
OAuth via hq.groupbuddies.com

### PR DESCRIPTION
**Pull request by naps62**
screenshot from 2015-02-11 18 06 29

@johnny1337 how bad does this look?

### CODE REVIEW

**Comment by joaojusto** [comment]
The tests are failing

**Comment by joaojusto** [comment]
The tests are failing

**Comment by johnny1337** [comment]
@naps62  The buttons are in a different style, that's the main problem for me.
Do we need the inputs at the bottom?

**Comment by naps62** [comment]
@johnny1337 Button fixed now (and this is a test to check if you actually get notifications now :monkey: )

@gabrielpoca @joaojusto is there anyone reviewing this repo, or should I just go away and merge?

**Comment by johnny1337** [comment]
@naps62  looks a lot better :+1:

File: config/autoprefixer.yml [hunk 1]
```diff
@@ -1,3 +1,2 @@
 browsers:
-  - "iOS 7"
```
Comment (joaojusto) [inline config/autoprefixer.yml]: Did you test if this still works on iOS without this line?

File: app/controllers/omniauth_callbacks_controller.rb [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+  skip_authorization_check
+
+  def headquarters
+    user = User.from_omniauth(auth_hash)
+
+    if user.persisted?
+      sign_in_and_redirect user, event: :authentication
+      set_flash_message(:notice, :success, kind: :headquarters) if is_navigational_format?
+    else
+      session['devise.headquarters_data'] = request.env['omniauth.auth']
```
Comment (gabrielpoca) [inline app/controllers/omniauth_callbacks_controller.rb]: Is this somehow related to the headquarters gem?

File: config/autoprefixer.yml (hunk 1 above)
Comment (naps62) [inline config/autoprefixer.yml]: this was not suposed to be deleted. fixing

File: app/controllers/omniauth_callbacks_controller.rb (hunk 2 above)
Comment (naps62) [inline app/controllers/omniauth_callbacks_controller.rb]: No, it's from devise. It was copied from their example in the docs. https://github.com/plataformatec/devise/wiki/OmniAuth:-Overview

File: app/models/user.rb [hunk 3]
```diff
@@ -12,4 +14,22 @@ class User < ActiveRecord::Base
   def display_name
     username.presence || "#{first_name} #{last_name[0]}"
   end
+
+  def name=(name)
+    self.first_name, self.last_name = name.split(' ', 2)
+  end
+
+  def self.from_omniauth(auth_hash)
```
Comment (gabrielpoca) [inline app/models/user.rb]: Have you considered extracting this public method from here? It doesn't seem to belong here.

Comment (naps62) [inline app/models/user.rb]: true. I'll extract it to the OAuth controller. It's only used there anyway

**Comment by gabrielpoca** [comment]
make the testes pass first, and then :ship:
